### PR TITLE
[vcpkg baseline][libtasn1] Fix static build

### PR DESCRIPTION
**Pull request by JackBoosY**
Fix static build on Windows:
```
   Creating library asn1Parser.lib and object asn1Parser.exp
asn1Parser.obj : error LNK2019: unresolved external symbol __imp_asn1_parser2tree referenced in function main
asn1Parser.obj : error LNK2019: unresolved external symbol __imp_asn1_parser2array referenced in function main
asn1Parser.obj : error LNK2019: unresolved external symbol __imp_asn1_delete_structure referenced in function main
asn1Parser.obj : error LNK2019: unresolved external symbol __imp_asn1_strerror referenced in function main
asn1Parser.exe : fatal error LNK1120: 4 unresolved externals
   Creating library asn1Coding.lib and object asn1Coding.exp
asn1Coding.obj : error LNK2019: unresolved external symbol __imp_asn1_parser2tree referenced in function main
asn1Coding.obj : error LNK2019: unresolved external symbol __imp_asn1_print_structure referenced in function main
asn1Coding.obj : error LNK2019: unresolved external symbol __imp_asn1_create_element referenced in function main
asn1Coding.obj : error LNK2019: unresolved external symbol __imp_asn1_delete_structure referenced in function main
asn1Coding.obj : error LNK2019: unresolved external symbol __imp_asn1_write_value referenced in function main
asn1Coding.obj : error LNK2019: unresolved external symbol __imp_asn1_der_coding referenced in function main
asn1Coding.obj : error LNK2019: unresolved external symbol __imp_asn1_strerror referenced in function main
...
```
Add `ASN1_STATIC` to `C_FLAGS` / `CXX_FLAGS` because this macro does not exist in the source configuration file.

Related: https://github.com/microsoft/vcpkg/pull/16953

### CODE REVIEW

File: ports/libtasn1/portfile.cmake [hunk 1]
```diff
@@ -29,6 +29,12 @@ else()
     set(VCPKG_CXX_FLAGS "-g -O2")
 endif()
 
+# The upstream doesn't add this macro to the configure
+if (VCPKG_TARGET_IS_WINDOWS AND VCPKG_LIBRARY_LINKAGE STREQUAL static)
+    set(VCPKG_C_FLAGS ${VCPKG_C_FLAGS} -DASN1_STATIC)
+    set(VCPKG_CXX_FLAGS ${VCPKG_CXX_FLAGS} -DASN1_STATIC)
```
Comment (dg0yt) [inline ports/libtasn1/portfile.cmake:35]: Overwrite for debug and release, not `string(APPEND ...)`?

Comment (JackBoosY) [reply ports/libtasn1/portfile.cmake:35]: It's same with `string(APPEND ...)`.

Comment (dg0yt) [reply ports/libtasn1/portfile.cmake:35]: But it is assigning 
`"${VCPKG_C_FLAGS};-DASN1_STATIC"` 
where you need to assign 
`"${VCPKG_C_FLAGS} -DASN1_STATIC"`.

**Review by PhoebeHui — APPROVED** [review]

File: ports/libtasn1/portfile.cmake (hunk 1 above)
Comment (PhoebeHui) [reply ports/libtasn1/portfile.cmake:35]: Could you patch the buildsystem directly?

File: versions/l-/libtasn1.json [hunk 2]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "4969c8ddb39ae78f59999f2e94e6f0c8d8e81aaa",
```
Comment (JackBoosY) [inline versions/l-/libtasn1.json:4]:
```suggestion
      "git-tree": "04828737d466a23ff937ef771c49e861ded218c2",
```

**Review by PhoebeHui — APPROVED** [review]
LGTM

File: ports/libtasn1/portfile.cmake [hunk 3]
```diff
@@ -29,6 +29,11 @@ else()
     set(VCPKG_CXX_FLAGS "-g -O2")
 endif()
 
+# The upstream doesn't add this macro to the configure
+if (VCPKG_TARGET_IS_WINDOWS AND VCPKG_LIBRARY_LINKAGE STREQUAL static)
```
Comment (dg0yt) [inline ports/libtasn1/portfile.cmake:33]: -Wpedantic:  ... `STREQUAL "static"`

Comment (JackBoosY) [inline ports/libtasn1/portfile.cmake:33]:
```suggestion
if (VCPKG_TARGET_IS_WINDOWS AND VCPKG_LIBRARY_LINKAGE STREQUAL "static")

```

File: versions/l-/libtasn1.json [hunk 4]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "04828737d466a23ff937ef771c49e861ded218c2",
```
Comment (JackBoosY) [inline versions/l-/libtasn1.json:4]:
```suggestion
      "git-tree": "3554f6c03cdac32ddf68540d62c04f6f4644ec94",
```

**Pull request merged**